src/portable/st/stm32_fsdev/dcd_stm32_fsdev.c

```diff
@@ -623,13 +622,15 @@ bool dcd_edpt_open (uint8_t rhport, tusb_desc_endpoint_t const * p_endpoint_desc
 
   if(dir == TUSB_DIR_IN)
   {
+    // FIXME: use pma_alloc to allocate memory dynamically
```

Comment:
I am not sure if it is possible to not use alloc memory for stm32fsdev. I absolutely don't want the dcd to have dynamic memory alloc even if that means we have to impose some limitation on the port.

Reply:
Actually, for the case that CAN is not being used and no double-bufferig, I think we can statically allocate 64 bytes per endpoint. This will be simpler for now. 

Note I wasn't going to use the system malloc. PMA is not normal RAM, it's a weird double-ported thing. I was going to write a custom allocator.

Reply:
@pigrew please use that static 64 bytes per endpoints. My will to resist any form of dynamic allocation is very strong :D. we can define an address that is the start of PMA used by USB, since it is static, application could figure out where is the end of PMA used by USB and use the rest for CAN. If needed user can modify it, or we can export the configuration via an OPT_ later on. We can always revisit this later if it is an issue.

Reply:
I'm going to second this... no dynamic allocation, I can just _see_ how this will cause all sorts of fun problems down the road.

Reply:
Removed in latest patch. Still untested. The F103 will now only support 4 bi-directional endpoints (4 IN and 4 OUT) by default, instead of the 8 it previously did.